glide-core/redis-rs/redis/src/cluster_async/mod.rs

```diff
@@ -1900,7 +1906,7 @@ where
         // Reset the current slot map and connection vector with the new ones
         let mut write_guard = inner.conn_lock.write().expect(MUTEX_WRITE_ERR);
         let read_from_replicas = inner
-            .get_cluster_param(|params| params.read_from_replicas)
+            .get_cluster_param(|params| params.read_from_replicas.clone())
             .expect(MUTEX_READ_ERR);
```

Comment:
I don't see that we are accessing `write_guard` in

I suggest to move this before we acquire the lock (this reduces the locking time).

Also, `MUTEX_READ_ERR` seems incorrect here

Reply:
My changes do not affect this. I added clone() because we do not have a Copy trait in String and we used it to save the Az.